docs/en/modules/ws2812.md

```diff
@@ -181,6 +181,87 @@ The number of given bytes must match the number of bytesPerLed of the buffer
 buffer:fill(0, 0, 0) -- fill the buffer with black for a RGB strip
 ```
 
+## ws2812.buffer:dump()
+Returns the contents of the buffer (the pixel values) as a string. This can then be saved to a file or sent over a network.
+
+#### Syntax
+`buffer:dump()`
+
+#### Returns
+A string containing the pixel values. 
+
+#### Example
+```lua
+local s = buffer:dump() 
+```
+
+## ws2812.buffer:load()
+Converts a string into a buffer (see also the `dump` method). The buffer must have the correct shape or an error will be thrown.
+
+#### Syntax
+`buffer:load(string)`
+
+#### Parameters
+ - `string` the pixel values to be set into the buffer
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+`buffer:load(anotherbuffer:dump()) -- copy a buffer
+```
+
+## ws2812.buffer:mix()
+This is a general method that loads data into a buffer that is a linear combination of data from other buffers. It can be used to copy a buffer or,
+more usefully, do a cross fade. The pixel values are computed as integers and then range limited to [0, 255]. This means that negative
+factors work as expected, and that the order of combining buffers does not matter.
+
+#### Syntax
+`buffer:mix(factor1, buffer1, ...)`
+
+#### Parameters
+ - `factor1` This is the factor that the contents of `buffer1` are multiplied by. This factor is scaled by a factor of 256. Thus `factor1` value of 256 is a factor of 1.0.
+ - `buffer1` This is the source buffer. It must be of the same shape as the destination buffer.
+
+There can be any number of factor/buffer pairs.
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+-- loads buffer with a crossfade between buffer1 and buffer2
+buffer:mix(256 - crossmix, buffer1, crossmix, buffer2)
+
+-- multiplies all values in buffer by 0.75
+-- This can be used in place of buffer:fade
+buffer:mix(192, buffer)
+```
+
+## ws2812.buffer:power()
+Computes the total energy requirement for the buffer. This is merely the total sum of all the pixel values (which assumes that each color in each
+pixel consumes the same amount of power).
```

Comment:
The WS2812 is just three leds with a WS2811 chip. The datasheet for this chip (e.g. https://cdn-shop.adafruit.com/datasheets/WS2811.pdf ) indicates that each channel (R, G, B) is a 20ma constant current drive. The brightness is controlled by a variable duty cycle (per channel). I *assumed* that the duty cycle scaled linearly with the pixel value (it would be more difficult to to otherwise). This means that the *average* current is colorvalue / 255 * 20mA.